src/pypromice/process/aws.py

```diff
@@ -38,15 +50,28 @@ def __init__(self, config_file, inpath, var_file=None, meta_file=None):
         meta_file: str, optional
             Metadata info file path. If not given then pypromice's
             metadata file is used. The default is None.
-        '''
-        assert(os.path.isfile(config_file)), "cannot find "+config_file
-        assert(os.path.isdir(inpath)), "cannot find "+inpath
-        logger.info('AWS object initialising...')
+        """
+        assert os.path.isfile(config_file), "cannot find " + config_file
+        assert os.path.isdir(inpath), "cannot find " + inpath
+        logger.info("AWS object initialising...")
 
         # Load config, variables CSF standards, and L0 files
         self.config = self.loadConfig(config_file, inpath)
-        self.vars = load.getVars(var_file)
-        self.meta = load.getMeta(meta_file)
+        self.vars = pypromice.resources.load_variables(var_file)
+        self.meta = pypromice.resources.load_metadata(meta_file)
+
+        config_hash = get_commit_hash_and_check_dirty(Path(config_file))
+        config_source_string = f"{Path(config_file).name}:{config_hash}"
+        inpath_hash = get_commit_hash_and_check_dirty(Path(inpath))
+        inpath_source_string = f"{Path(inpath).name}:{inpath_hash}"
+
+        source_dict = dict(
+            pypromice = metadata.version("pypromice"),
+            l0_config_file = config_source_string,
+            l0_data_root = inpath_source_string,
+        )
+        self.meta["source"] = json.dumps(source_dict)
```

Comment:
I agree. We should also include the data issues repository. I forgot.

Reply:
Yes. It will be better to provide the data-issues repository as an input parameter similar to aws-l0, where the path and thereby the version is deterministic.